torch/_inductor/compile_fx.py

```diff
@@ -219,88 +224,630 @@ def remove_unaligned_input_idxs(inputs, static_input_idxs):
     return static_input_idxs
 
 
-def cudagraphify_impl(model, inputs, static_input_idxs=()):
+@dataclasses.dataclass(frozen=True)
```

Comment:
Bah, does this really have to live in torch._inductor? This should work for any backend right?

Reply:
Yea, can move out in the next pr

Reply:
In fact, the CUDAGraphTape isn't even a compiler only abstraction; the eager mode cuda graphs api could hypothetically use this too.

Reply:
You would need to add the same guards that exist in dynamo, but otherwise, yes.